src/Orleans.Runtime/Hosting/GenericHostExtensions.cs

```diff
@@ -12,6 +11,44 @@ namespace Microsoft.Extensions.Hosting
     /// </summary>
     public static class GenericHostExtensions
     {
+        /// <summary>
+        /// Configures the host app builder to host an Orleans silo.
+        /// </summary>
+        /// <param name="hostAppBuilder">The host app builder.</param>
+        /// <returns>The host builder.</returns>
+        public static HostApplicationBuilder UseOrleans(
+            this HostApplicationBuilder hostAppBuilder) =>
+            hostAppBuilder.UseOrleans(_ => { });
+
+        /// <summary>
+        /// Configures the host builder to host an Orleans silo.
+        /// </summary>
+        /// <param name="hostAppBuilder">The host app builder.</param>
+        /// <param name="configureDelegate">The delegate used to configure the silo.</param>
+        /// <returns>The host builder.</returns>
+        /// <remarks>
+        /// Calling this method multiple times on the same <see cref="HostApplicationBuilder"/> instance will result in one silo being configured.
+        /// However, the effects of <paramref name="configureDelegate"/> will be applied once for each call.
+        /// </remarks>
+        public static HostApplicationBuilder UseOrleans(
+            this HostApplicationBuilder hostAppBuilder,
+            Action<ISiloBuilder> configureDelegate)
+        {
+            ArgumentNullException.ThrowIfNull(hostAppBuilder);
+            ArgumentNullException.ThrowIfNull(configureDelegate);
+
+            if (hostAppBuilder.Configuration["HasOrleansClientBuilder"] is not null)
```

Comment:
I think this should use `Properties` instead of `Configuration`

Reply:
`Properties` doesn't exist on this API.